lib/hocon/impl/simple_config_list.rb

```diff
@@ -149,11 +216,122 @@ def render_value_to_sb(sb, indent_size, at_root, options)
     end
   end
 
-  def new_copy(origin)
-    Hocon::Impl::SimpleConfigList.new(origin, @value)
+  def contains?(o)
+    value.include?(o)
   end
 
   def include_all?(value_list)
     value_list.all? { |v| @value.include?(v)}
   end
+
+  def contains_all?(c)
+    include_all?(c)
+  end
+
+  def get(index)
+    value[index]
+  end
+
+  def index_of(o)
+    value.index(o)
+  end
+
+  def is_empty
+    empty?
+  end
+
+  # Skipping upstream definition of "iterator", because that's not really a thing
+  # in Ruby.
+
+  def last_index_of(o)
+    value.rindex(o)
+  end
+
+  # skipping upstream definitions of "wrapListIterator", "listIterator", and
+  # "listIterator(int)", because those don't really apply in Ruby.
+
+  def sub_list(from_index, to_index)
+    value[from_index..to_index]
+  end
+
+  def to_array
+    value
+  end
+
+  def we_are_immutable(method)
+    ConfigBugOrBrokenError.new("ConfigList is immutable, you can't call List. '#{method}'")
```

Comment:
I'm wondering if we should throw a different exception here. The Java throws `UnsupportedOperationException`, which doesn't exist in Ruby, but in the upstream library a `ConfigBugOrBrokenException` is only thrown in cases where something is broken in the library itself rather than in cases where the user has broken something or attempted something that's disallowed, and I think we should follow that pattern.